### DISCUSSION TITLE
Schedule Job for Netbox Devices

### DISCUSSION THREAD

**Discussion by Nino-Rey** [netbox-community/netbox#9720] [Q&A]
Hi everyone! I just want to ask if there's a way I can run a schedule job, let's say weekly  exporting Netbox > Devices. My plan is I will upload the exported CSV file to S3 bucket.

**Comment by candlerb** [comment]
Netbox itself doesn't do scheduled tasks, but you can write a shell or python script and run it from cron - and your script can also do the upload to S3.

Beware that a CSV export of devices cannot be faithfully re-imported back into Netbox.

Specifically: you'll be able to import *some* of the fields but not all of them (e.g. you can't import primary IP address, because you first have to create the interfaces and IP addresses).  And you won't be able to restore the device ID.

If all you want is a manual record of what the device inventory looked like at a point in time, it'll be OK.  Just don't treat it as a backup, because if your old Netbox instance dies, you definitely won't be able to recreate it from CSV exports.

**Comment by ryanmerolle** [comment]
#8366 is hopefully slated for the 3.4 release at the end of the year, but could slip.
